pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -273,31 +273,35 @@ func fetchTeamByName(workflowCtx *workflow.Context, teamName string) (*mongodbat
 		return nil, err
 	}
 
-	return atlasTeam, nil
+	return atlasTeamResponse, nil
 }
 
-func createTeam(workflowCtx *workflow.Context, atlasTeam *mongodbatlas.Team) (*mongodbatlas.Team, error) {
+func createTeam(workflowCtx *workflow.Context, atlasTeam *admin.Team) (*admin.TeamResponse, error) {
 	workflowCtx.Log.Debugf("create team named %s in atlas", atlasTeam.Name)
-	atlasTeam, _, err := workflowCtx.Client.Teams.Create(workflowCtx.Context, workflowCtx.OrgID, atlasTeam)
+	atlasTeam, _, err := workflowCtx.SdkClient.TeamsApi.CreateTeam(workflowCtx.Context, workflowCtx.OrgID, atlasTeam).Execute()
 	if err != nil {
 		return nil, err
 	}
-
-	return atlasTeam, nil
+	response := &admin.TeamResponse{}
```

Comment:
this converts an internal SDK type to another internal SDK type. Let's try to avoid this. Rather, just return `atlasTeam` from the `CreateTeam` invocation directly.